mirascope/core/groq/stream.py

```diff
@@ -49,3 +56,55 @@ def _construct_message_param(
         if tool_calls:
             message_param["tool_calls"] = tool_calls
         return message_param
+
+    def construct_call_response(self) -> GroqCallResponse:
+        """Constructs the call response from a consumed GroqStream."""
+        if self.message_param is None:
+            raise ValueError(  # pragma: no cover
+                "No stream response, check if the stream has been consumed."
+            )
+        message = {
+            "role": self.message_param["role"],
+            "content": self.message_param.get("content", ""),
+            "tool_calls": self.message_param.get("tool_calls", []),
+        }
+        usage = CompletionUsage(
```

Comment:
usage can be `None` and we should handle that so we get `None` cost instead of `0` in that case

Reply:
done